mw-webapp/src/hooks/useLoad.ts

```diff
@@ -0,0 +1,75 @@
+import {useEffect, useState} from "react";
+
+/**
+ * Use Load hook Props
+ */
+interface useLoadProps<T, D> {
+
+  /**
+   * Callback that is called to fetch data
+   */
+  loadData: () => Promise<T>;
+
+  /**
+   * Callback that is called to validate data
+   */
+  validateData?: (data: T) => boolean;
+
+  /**
+   * Callback that is called on fetch and validation success
+   */
+  onSuccess: (data: T) => void;
+
+  /**
+   * Callback this is called on fetch or validation error
+   */
+  onError?: (error: Error) => void;
+
+  /**
+   * Passed dependency to refetch and revalidate data
+   */
+  dependency?: D;
+}
+
+/**
+ * Custom hook to fetch, validate, and manage the state of data
+ */
+export function useLoad<T, D>({
+  loadData,
+  validateData = () => true,
+  onSuccess,
+  onError = () => {},
+  dependency = undefined,
+}: useLoadProps<T, D>) {
+  const [data, setData] = useState<T>();
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+
+    /**
+     * Fetches data using loadData callback, validates it, then updates state if fetch was successful
+     */
+    async function fetchAndValidateData() {
+      try {
+        setIsLoading(true);
+        const loadedData = await loadData();
+        if (!validateData(loadedData)) {
+          throw new Error("Data validation failed");
+        }
+
+        setData(loadedData);
+        onSuccess(loadedData);
+      } catch (err) {
+        if (err instanceof Error) {
```

Comment:
Could you describe the case when error is not instance of Error and why we should do nothing in this case?

Reply:
I don't know much about other possible Errors, i think it can be "FirebaseError" too, don't really know how to handle everyting

Reply:
- But catch block catches all errors in js by default. So why do you want to handle only one type and why do you want to ignore errors like "FirebaseErrors"

Reply:
I found that FirebaseError have instance of Error, so it will work

Reply:
We need to use instanceof to convert unknown into Error type, we can't pass it into onError function if we don't do it

Reply:
So I still don't understand why should not you handle errors if they have other type, could you explain it please?

Reply:
@tritonJS826 Any error that can occur have instance of Error, because they are all based on it, FirebaseError = Error + additional properties

Reply:
ok, if it is really safe